composer/lib/dependabot/composer/update_checker.rb

```diff
@@ -35,6 +35,12 @@ def latest_resolvable_version
             latest_allowable_version: latest_version_from_registry,
             requirements_to_unlock: :own
           ).latest_resolvable_version
+
+        unless allowed_versions.include?(@latest_resolvable_version)
```

Comment:
This doesn't seem right to me, `latest_version_from_registry` which we pass into `VersionResolver` should _already_ exclude any ignored versions, because `VersionResolver#latest_version` already filters that out.

If for some reason `VersionResolver` is still returning ignored versions, we should pass the `ignored_versions` into that class as well, and make sure it handles them correctly